### DISCUSSION TITLE
Colour scheme using ANSI basic colours to match terminal theme

### DISCUSSION THREAD

**Discussion by lachsh** [jotaen/klog#307] [Feature ideas & discussions]
Hello!

It would be great to get a mode/setting to get all output to use only the *basic* ANSI colour escapes i.e. the first section shown here: [https://talyian.github.io/ansicolors/](https://talyian.github.io/ansicolors/)

I saw the option to use a light theme exists, which can help with some terminal configurations, but using combinations of the 8 basic colours + bold/dim/etc modifiers means the output will match the colour theme of the whole terminal on almost every terminal emulator.

This helps with both :sparkles: aesthetics :sparkles: (I like to have a pretty terminal 🎨) and also readability, as with my present terminal theme, not all of the colours used in either the built-in dark or light theme are 100% readable.

I'd almost say there's an argument to be made that sticking to the base 16 colours should be the default and more advanced light and dark themes should be explicitly enabled using the config, but I would also be completely happy keeping things as-is with a config option to use the basic colours.

Thank you :)

**Comment by jotaen** [comment]
Thanks for bringing this up. I’d probably want to stick with the current defaults, but I think it wouldn’t harm to add another colour theme such as `basic` or `simple`.

**Comment by vladdeSV** [comment]
Just want to chime in that if klog was using the default ANSI colors, there would not be any need for dark or light color schemes – the same colors would always be used and work perfectly, and it would look much better ✨ if the colors matched my terminal palette.

**Reply by vladdeSV** [reply]
I printed one of my work days, and then manually recreated it with the usual ANSI colors… and I might have to take back that an explicit light or dark mode is not required. On light themes, the date and time spans become super hard to read no matter which light theme I used 🙁.

My own custom theme:

<table>
<tr>
<td>

<img width="778" alt="Screenshot 2024-08-03 at 12 20 45" src="https://github.com/user-attachments/assets/33f93e51-b31c-4dc6-88f9-8d8780ef15bb">

</td>
<td>

<img width="778" alt="Screenshot 2024-08-03 at 12 21 00" src="https://github.com/user-attachments/assets/fe85f920-20f6-4c71-96c9-720802b305fe">

</td>
</tr>
</table>


iTerm2's default "High Contrast" theme:
<table>
<tr>
<td>

<img width="778" alt="Screenshot 2024-08-03 at 12 21 26" src="https://github.com/user-attachments/assets/01e3cc2d-75d6-4fe7-a6d0-26d0a063f8d3">

</td>

<td>

<img width="778" alt="Screenshot 2024-08-03 at 12 21 15" src="https://github.com/user-attachments/assets/392054ca-b119-49bd-a895-3c4ce1af64d8">

</td>
</tr>
</table>

iTerm2's default "Solarized" theme:
<table>
<tr>
<td>

<img width="778" alt="Screenshot 2024-08-03 at 12 26 03" src="https://github.com/user-attachments/assets/25820b28-cf56-4a20-aad9-a6e3fc397030">

</td>

<td>

<img width="778" alt="Screenshot 2024-08-03 at 12 26 10" src="https://github.com/user-attachments/assets/35e2da93-04fd-475d-8acc-64a4da740798">

</td>
</tr>
</table>

**Reply by jotaen** [reply]
But wouldn’t you adjust the ANSI colours anyway? I.e., if you switch to a light terminal theme, wouldn’t you tune the colours so that they are easy to read?

By the way, I too noticed that on my terminal (Terminal.app, the MacOS default one), the preconfigured ANSI colours of most built-in themes are not well readable. I think this was also a reason back then for me to provide custom 256 bit colours for the output. I still think there is value in offering a basic 8-bit colour theme via the `config.ini` (https://github.com/jotaen/klog/issues/325).

Another, rather fancy idea for the longer run could be to provide an ability to completely customise the output colouring of klog. Might be overkill, but could be fun.

**Reply by vladdeSV** [reply]
> […] wouldn’t you tune the colours so that they are easy to read?

In the best of worlds that would be the case, but unfortunately that rarely works 😕 – most light themes' colors I've seen are hard to read on the default background (like iTerm2's default "High Contrast" color palette)

One example I can think of is Vim, which changes which colors it defaults to when you specify the theme of the terminal; `:set bg=light` or `:set bg=dark`

**Reply by vladdeSV** [reply]
I'm gonna backtrack (again 🙈), and say that a light / dark mode probably is not needed – it really should be up to the color palette to be readable. (I ended up changing the colors of my own custom theme).

Solarized above has readable time spans (eg. `8:00 - 12:00`), and if black and white colors are not used, I believe it should be readable no matter what.

My suggestion would be to use normal foreground color for the dates (eg. `2024-08-01`), and it should all be fine 😊

**Comment by jotaen** [comment]
I’ve created ~https://github.com/jotaen/klog/issues/325~ https://github.com/jotaen/klog/pull/328.

**Reply by jotaen** [reply]
The new `basic` colour scheme has been merged and will be available as of the next release.

**Reply by ccjmne** [reply]
Thank you very much, @jotaen, @vladdeSV.
It's a bit of a useless ping but, considering how little interactions happen on here, I thought you might like to know that at least one other user is really quite happy with the `basic` colour scheme!

**Reply by lx4r** [reply]
Finally I can configure klog so that the output has proper contrast in my terminal. Thank you, @jotaen and @vladdeSV ❤️ 
A bit of feedback:
- `klog config` just says "The config property must be one of: `dark`, `light`, `basic`, or `no_colour`". Without finding this discussion, I would've never guessed what effect `basic` has. How about adding a short explanation?
- The link to the issue above seems to be broken.

**Reply by jotaen** [reply]
> klog config just says "The config property must be one of: dark, light, basic, or no_colour". Without finding this discussion, I would've never guessed what effect basic has. How about adding a short explanation?

Good idea, that sounds reasonable. Would you have a suggestion in mind?

> The link to the issue above seems to be broken.

The respective PR was https://github.com/jotaen/klog/pull/328.

I’ve recently deactivated issues on this repo altogether, in favour of only using the Github discussions feature. Many of the issues, such as the one linked above, directly originated in a discussion anyway and didn’t contain any further information on their own. (Any other “original” issues were converted to a discussion after the fact.)
